Massive Overhaul

benhills opened the pull request:
I wanted to break out some of the functions so that it is more clear where people can plug in. I also moved all the variables into a dictionary so that we are not importing and exporting so much from each function. We talked about using a Pandas dataframe... I think the best way to do this if we wanted to be more 'proper' would be to make a data class and put all the variables as well as most of the functions inside of that (low priority for now though).

I feel like I am getting closer to the point that hopefully others can hop in and make some figures. 

Grace - Before we merge this branch, can you get onto devel and make sure that it works for you?

Edit: Waiting on #19 before merging this.

benhills committed:
Create an Uncertainty Notebook

Starting a notebook for uncertainty analysis. For now, we are only to
the point that we have a plot of correlation vs. offset for one subset
of one profile.

benhills committed:
Move beam loop into function

The velocity calculation was being done for all beams in a loop within
the calculate function. Trying to clean this up.

benhills committed:
Doing an overhaul of the data structuring

There are way too many separate arrays. I am moving them all into a
dictionary for now. It might be better to restructure this as a class
eventually, but it might be too big a step for the group right now.

benhills committed:
Massive restructuring

Separating processing functions and limiting the number of variables
that we are using.

benhills committed:
Changes to the introduction tutorial

Probably at the stage that I can have Grace look at it.

barcheck commented:
Looks great, Ben. Looks like you also added a step to interpolate places where there are nans? I think we don't want to interpolate nans linearly, b/c lines have unexpected behavior in the correlation functions - ex., a flat line correlated with a flat line might be highly correlated, despite there being no signal. We'll have to think more about how to fill in gaps / if we need to.

barcheck commented:
oh, also, we need to retain options for the 'filt' function

benhills commented:
Ok. I will get on both of these points. Thanks for looking at it. 

I am going to try to do this today and then I can start making some figures.

benhills commented:
So, the options for the filt function are still there, they just default to the values that we had been using a lot. If you want to change them though, they still exist. 

In regard to the nan interpolation, I just copied that from the loop. I think it was being done in previous versions, maybe that was unintentional though. Anyway, it is not hardcoded into the velocity function, just an additional function. Maybe I will add a warning to that funciton.

benhills committed:
Add some function descriptions

benhills committed:
Merge branch 'devel' into uncertainty

Bringing in the reorganization changes that were made.

benhills committed:
Update variable names to be more streamlined

We were keeping track of a bunch of different variables like 'h_full',
etc. I just streamlined this but added a flags field so that we can tell
what processing steps have been done.

benhills committed:
Small changes to the multi-beam plotting script

I wanted to remove the dependency on the MOA file.

benhills committed:
Remove this old script with an unused function in it.

benhills committed:
Clean up the function for extracting Measures velocities.

benhills committed:
Change the vector rotation for Measures velocities

I think that the conversion was ignoring one of the velocity components.
I tested this updated conversion for all combinations of +/- vx and vy
terms.

benhills commented:
Wait on #19 before merging this.

barcheck committed:
re-arrange correlation steps slightly; add function to plot a single correlation function; update tutorial notebook to reflect

RGFell committed:
add a line on the documentation and cp old functionality but commented#

benhills committed:
Add loop through all beams for Measures extraction.

benhills committed:
Array of track angles instead of using the first.

In converting from Measures Velocity (x/y) to along/across we were
originally using only the first angle. However, the angle changes over
the length of the track, so calculating a full array is preferred.

benhills committed:
Very minor changes to clean up the PR

benhills committed:
Merge pull request #20 from ICESAT-2HackWeek/corr_plots

re-arrange correlation steps slightly; add function to plot a single …

benhills committed:
Merge branch 'devel' into get_measures

benhills committed:
Update the flip_meas_sign arg in plotting function

RGFell committed:
Merge pull request #19 from ICESAT-2HackWeek/get_measures

Get Measures Velocities
